### ISSUE TITLE
[BUG - Develop] Can't `|` two SaverDicts

### ISSUE THREAD

**Issue by InspectorCaracal** [evennia/evennia#2789]
#### Describe the bug
When combining two attributes containing dict data, it fails with a traceback.
```
File "./TestGame/typeclasses/characters.py", line 30, in test_attr
return self.db.db_one | self.db.db_two
File "./evennia/evennia/utils/dbserialize.py", line 243, in __or__
return self._data | other
TypeError: unsupported operand type(s) for |: 'dict' and '_SaverDict
```

#### To Reproduce
1. Store dicts in two attributes or attribute properties.
2. Use the `|` operator on them

#### Develop-branch commit
22fa2c6b8

**Comment by Griatch** [comment]
Well, this one's easy - they just don't have an operator for using | (which feels pretty esoteric to do, honestly).

**Comment by Griatch** [comment]
Ok, esoteric was the wrong term, but  it has at least not been a thing until Python 3.9+, which is why it was not originally in there. 🙂

**Comment by InspectorCaracal** [comment]
Oh haha I completely forgot it was so recent! I hadn't really ever needed to combine dicts until I started messing around with splicing together prototype dicts in evennia, so it's the syntax I learned first for it.

Anyway, the main issue is that it works fine if you combine a normal dict and a saver dict, but breaks if you use two saver dicts. It's easy enough to work around by casting one or both to `dict` first so I don't think it's a priority, but since it tracebacked I figured I'd report it.

**Comment by Griatch** [comment]
Yeah, the 'traditional' way, pre-Python3.9 of doing this is `dict1.update(dict2)`.
But yeah, it should be added of course.

**Comment by aogier** [comment]
Being a reflected dunder fan, I've published a [trivial fix](https://github.com/evennia/evennia/pull/2813) for this!

**Comment by Griatch** [comment]
Resolved in #2813 !
